packages/hash/api/src/model/comment.model.ts

```diff
@@ -0,0 +1,211 @@
+import { TextToken } from "@hashintel/hash-shared/graphql/types";
+import {
+  Entity,
+  EntityConstructorArgs,
+  EntityExternalResolvers,
+  Comment,
+  UnresolvedGQLEntityType,
+  User,
+  Block,
+} from ".";
+import { DbClient } from "../db";
+import {
+  DbCommentProperties,
+  DbTextProperties,
+  EntityType,
+} from "../db/adapter";
+import {
+  SystemTypeName,
+  Comment as GQLComment,
+  LinkedEntityDefinition,
+} from "../graphql/apiTypes.gen";
+
+export type CommentExternalResolvers = EntityExternalResolvers | "contents"; // contents resolved in `src/graphql/resolvers/Comments/linkedEntities.ts`
+
+export type UnresolvedGQLComment = Omit<
+  GQLComment,
+  CommentExternalResolvers
+> & {
+  entityType: UnresolvedGQLEntityType;
+};
+
+export type CommentConstructorArgs = {
+  properties: DbCommentProperties;
+} & EntityConstructorArgs;
+
+class __Comment extends Entity {
+  properties: DbCommentProperties;
+
+  constructor(args: CommentConstructorArgs) {
+    super(args);
+    this.properties = args.properties;
+  }
+
+  static async getEntityType(client: DbClient): Promise<EntityType> {
+    const commentEntityType = await client.getSystemTypeLatestVersion({
+      systemTypeName: "Comment",
+    });
+    return commentEntityType;
+  }
+
+  static async createComment(
+    client: DbClient,
+    params: {
+      createdBy: User;
+      parent: Block;
+      accountId: string;
+      content: TextToken[];
+    },
+  ): Promise<Comment> {
+    const { createdBy, parent, accountId, content } = params;
+
+    const textProperties: DbTextProperties = {
+      tokens: content,
+    };
+
+    const linkedEntities: LinkedEntityDefinition[] = [
+      {
+        path: "$.contents",
+        destinationAccountId: accountId,
+        entity: {
+          entityType: {
+            systemTypeName: SystemTypeName.Text,
+          },
+          entityProperties: textProperties,
+        },
+      },
+      {
+        path: "$.owner",
+        destinationAccountId: accountId,
+        entity: {
+          existingEntity: {
+            accountId: createdBy.accountId,
+            entityId: createdBy.entityId,
+          },
+        },
+      },
```

Comment:
**Blocking:** why does this need to be captured in a link? How is it different from the the `comment.accountId` field?

If there's no difference I think we should rely on the `comment.accountId` field instead.

Reply:
The accountId of the comment is the id of the user the block belongs to. The user that created the comment is captured in the link. Reason: https://www.notion.so/Comments-Development-b805a75ac7d448feae01517c09ca2ccd?d=f8bbcf39efd04643ad0cb06b80a9e099#7f356352f6fd4f339999830e001b37bd

Reply:
Thanks for the link. I think `author` would be a more appropriate name for this link in that case, as we use `accountId` to determine who "owns" something - and in this scenario they would point to different things.

Reply:
Changed in 57f13998b702593c8e44e5e4332d06bf05d74eb2